File: src/main/java/dev/vrba/slack/tenor/bot/service/slack/dto/SlackViewOpenRequest.java
package dev.vrba.slack.tenor.bot.service.slack.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import io.micronaut.core.annotation.NonNull;
import io.micronaut.serde.annotation.Serdeable;
import lombok.AllArgsConstructor;
import lombok.Getter;

@Getter
@Serdeable
@AllArgsConstructor
public class SlackViewOpenRequest {
    @NonNull
    @JsonProperty("trigger_id")
    private final String trigger;

    @NonNull
    @JsonProperty("view")
    private final SlackView view;
}

File: src/main/kotlin/dev/vrba/slack/service/slack/dto/SlackViewOpenRequest.kt
package dev.vrba.slack.service.slack.dto

import com.fasterxml.jackson.annotation.JsonProperty
import io.micronaut.serde.annotation.Serdeable

@Serdeable
data class SlackViewOpenRequest(
    @JsonProperty("trigger_id")
    val trigger: String,
    @JsonProperty("view")
    val view: SlackView,
)
